cogs/afterhours/afterhours.py

```diff
@@ -54,32 +92,121 @@ def cog_unload(self):
+            self.logger.debug("Checking guild %s", guild.id)
+            async with self.config.guild(guild).get_attr(KEY_CHANNEL_IDS)() as channels:
+                staleIds = []
+                for channelId, data in channels.items():
+                    self.logger.debug("Checking channel ID %s", channelId)
+                    channel = discord.utils.get(guild.channels, id=int(channelId))
+                    if not channel:
+                        self.logger.error("Channel ID %s doesn't exist!", channelId)
+                        staleIds.append(channelId)
+                        continue
+                    creationTime = datetime.fromtimestamp(data["time"])
+                    self.logger.debug("Time difference = %s", datetime.now() - creationTime)
+                    if datetime.now() - creationTime > timedelta(seconds=DELETE_TIME):
+                        self.logger.info("Deleting channel %s (%s)", channel.name, channel.id)
+                        await channel.delete(reason="AfterHours purge")
+                        # Don't delete the ID here, this will be taken care of in
+                        # the delete listener
+                for channelId in staleIds:
+                    self.logger.info("Purging stale channel ID %s", channelId)
+                    del channels[channelId]
+
+    async def doAutoPurge(self, forced=False):
+        for guild in self.bot.guilds:
+            guildConfig: Group = self.config.guild(guild)
+            autoPurgeConfig: Group = guildConfig.get_attr(KEY_AUTO_PURGE)
+
+            if not forced and await autoPurgeConfig.get_attr(KEY_BACKGROUND_LOOP)() is False:
+                self.logger.debug(
+                    "Background execution of auto-purged is disabled for guild %s", guild.id
+                )
+                continue
+
+            # checking if the AfterHours role exists in this guild
+            ahRoleId: int = await guildConfig.get_attr(KEY_ROLE_ID)()
+            if not ahRoleId:
+                self.logger.debug("No AfterHours role ID set for guild %s", guild.id)
+            else:
+                ahRole: discord.Role = discord.utils.get(guild.roles, id=int(ahRoleId))
+                if not ahRole:
+                    self.logger.debug("AfterHours role does not exist in guild %s!", guild.id)
+
+            # a list of members to be purged
+            inactiveMembers: List[discord.Member] = []
+
+            # check for inactive members based on a set inactive duration
+            (
+                years,
+                months,
+                weeks,
+                days,
+                hours,
+                minutes,
+                seconds,
+            ) = await self.getAutoPurgeInactiveDuration(guild)
+
+            inactiveDurationTimeDelta = timedelta(
+                days=years * 365 + months * 30 + weeks * 7 + days,
+                hours=hours,
+                minutes=minutes,
+                seconds=seconds,
+            )
```

Comment:
Storing multiple ones was the intuitive way I came up with, because what we see when we input is what we also see when it gets output. However, after some thoughts, storing only seconds has more pluses than minuses compared to storing multiple units. I will incorporate the change in future commits.